### PR TITLE
utils: use clone_git_repo from osbs-client

### PR DESCRIPTION
**Pull request by mlangsdorf**
Use the clone_git_repo function from osbs-client instead of having
a separate but nearly identical version in atomic_reactor.

Depends on https://github.com/projectatomic/osbs-client/pull/846

Signed-off-by: Mark Langsdorf <mlangsdo@redhat.com>

### CODE REVIEW

File: atomic_reactor/core.py [hunk 1]
```diff
@@ -38,8 +38,8 @@
         BUILD_JSON, DOCKER_SOCKET_PATH, DOCKER_MAX_RETRIES, DOCKER_BACKOFF_FACTOR,\
         DOCKER_CLIENT_STATUS_RETRY
 from atomic_reactor.source import get_source_instance_for
-from atomic_reactor.util import (
-    ImageName, clone_git_repo, figure_out_build_file, Dockercfg)
+from atomic_reactor.util import (ImageName, figure_out_build_file, Dockercfg)
```
Comment (athos-ribeiro) [inline atomic_reactor/core.py]: Now this is in a single line, we could remove the parenthesis

Comment (mlangsdorf) [reply atomic_reactor/core.py]: Fixed.

File: atomic_reactor/util.py [hunk 2]
```diff
@@ -405,7 +351,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 shutil.rmtree(self.our_tmpdir)
 
     def reset(self, git_reference):
-        self._commit_id = reset_git_repo(self.git_path, git_reference)
+        self._commit_id, _ = reset_git_repo(self.git_path, git_reference)
```
Comment (rcerven) [inline atomic_reactor/util.py]: pass self._git_depth to reset_git_repo ?

Comment (mlangsdorf) [reply atomic_reactor/util.py]: That functionality is added in #1158

Comment (rcerven) [reply atomic_reactor/util.py]: ohhh ic

**Comment by mlangsdorf** [comment]
Not adding anything to RN since this is an internal only code organization change.

**Pull request merged**